code/harness/DiskContents.cpp

```diff
@@ -131,9 +114,9 @@ DiskContents::DiskContents(char* path, const char* type) {
 }
 
 DiskContents::~DiskContents() {
-  // free(disk_path);
-  // free(mount_point);
-  // free(fs_type);
+  free(disk_path);
+  free(mount_point);
+  free(fs_type);
 }
 
 int DiskContents::mount_disk() {
```

Comment:
nit: I would use string manipulations to build mount point

Reply:
I cannot comment on the proper line, but the do-while loop below to unmount the file system is not correct. You have a `system(umount)` call and then you enter the do-while loop which calls `umount()` until it succeeds. You should either remove the `system(umount)` call, or remove the `umount()` call and move the `system(umount)` call into the do-while loop.